lib/js-yaml/loader.js

```diff
@@ -289,6 +289,11 @@ function storeMappingPair(state, _result, overridableKeys, keyTag, keyNode, valu
   // We need to convert keyNode to a string, but doing so can hang the process
   // (deeply nested arrays that explode exponentially using aliases) or execute
   // code via toString.
+
+  if (typeof keyNode === 'object') {
+    throw new Error('Key is an object');
```

Comment:
Might want a slightly more useful error message to point out the problem. Can use `JSON.stringify(keyNode)` to convert the object to a string.

`throw new Error('Keys are not allowed to be objects, key: ' + JSON.stringify(keyNode));`